compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal

```diff
@@ -436,7 +423,7 @@ function createObjectField(jvm:MethodVisitor mv, bir:BObjectField field) {
     // Load flags
     // TODO: get the flags
     int visibility = 1;
-    if (field.visibility == "PACKAGE_PRIVATE") {
+    if (field.flags == 0) {
```

Comment:
This need to be changed like (mask & flag) == flag

Reply:
Fixed